v3/lint/base.go

```diff
@@ -219,24 +222,12 @@ func (l *CertificateLint) Execute(cert *x509.Certificate, config Configuration)
 	if l.Source == CABFBaselineRequirements && !util.IsServerAuthCert(cert) {
 		return &LintResult{Status: NA}
 	}
-	return l.execute(l.Lint(), cert, config)
-}
-
-func (l *CertificateLint) execute(lint LintInterface, cert *x509.Certificate, config Configuration) *LintResult {
-	configurable, ok := lint.(Configurable)
-	if ok {
-		err := config.Configure(configurable.Configure(), l.Name)
-		if err != nil {
-			details := fmt.Sprintf(
-				"A fatal error occurred while attempting to configure %s. Please visit the [%s] section of "+
-					"your provided configuration and compare it with the output of `zlint -exampleConfig`. Error: %s",
-				l.Name,
-				l.Name,
-				err.Error())
-			return &LintResult{
-				Status:  Fatal,
-				Details: details}
-		}
+	lint := l.Lint()
```

Comment:
Similar to the above centralization of `CheckApplies`,  lifted the interface cast-and-check into `Configuration` type itself so as to centralize the check (plus the error string construction).